mock/py/mockbuild/buildroot.py

```diff
@@ -359,6 +359,21 @@ def _init_aux_files(self):
                 with open(p, 'w+') as fo:
                     fo.write(chroot_file_contents[key])
 
+    @traceLog()
+    def ensure_rpm_config_home(self):
+        """Create a fake home directory with an appropriate .rpmmacros"""
+
+        rpm_config_home = os.path.join(self.basedir, 'rpmconfig')
+        util.mkdirIfAbsent(rpm_config_home)
+
+        # Since /proc and /sys are mounted special filesystems when RPM is running
+        # to install the buildroot, it doesn't make sense for RPM to try and
+        # set the permissions on them - and that might fail with permission errors.
+        with open(os.path.join(rpm_config_home, ".rpmmacros"), "w") as f:
+            f.write("%_netsharedpath /proc:/sys\n")
```

Comment:
Hmm, why this way? Isn't it better to set `config_opts['macros']['%_netsharedpath']` as default in util.py? And mention it in sitedefault.cfg.

Reply:
I'm probably missing something, but my impression was that config_opts['macros'] affects rpmbuild, but would have no effect on the package management commands used to set up the chroot. [Except for microdnf? Perhaps something is needed for that case as well?]

Reply:
AFAICS this is pretty much the only way to achieve the desired result at the moment if yum/dnf/whatever is run from outside the chroot. Rpm reads its configuration during Python module initialization, and since neither yum or dnf have a way to define arbitrary rpm macros via their configuration or cli-switches, this has to be placed in a file before yum/dnf is launched. If yum/dnf was run from within the target chroot then you could place the configuration there, but from the outside... you can't go changing system-wide configuration even temporarily, so that basically leaves what we have here: override HOME to a temporary directory and create ~/.rpmmacros there.

On the positive side, this could be used to house other similar configurations if needed, for example %_install_langs could be supported this way.

One possibility would be taking this all one step further and generate the temporary ~/.rpmmacros from config_opts['macros'] contents so the same config mechanism could be used for arbitrary macros on the target.

Reply:
I filed DNF RFE and discussed that with DNF team in person:
  https://bugzilla.redhat.com/show_bug.cgi?id=1673333
But it is a long run. It will not help with this PR - we can just make it easier later.

Reply:
Hmm. Just realized/remembered that while module import still always reads in the rpm config, for many years now it's possible to reset + reload the configuration later. So there is an alternative: doing rpm.reloadConfig() inside the chroot. Just FWIW.

Reply:
I am still not sure about that `.rpmmacros`. It seems very hackish to me. It is not needed on bare metals hosts. Only in containers. Therefore, I would rather see this done in Toolbox itself or in Container kickstarts https://pagure.io/fedora-kickstarts/tree/master

Reply:
Yes, this could be configured in /etc/rpm in the toolbox-support package which is suppose to be installed in all containers for toolbox, and I think it would get picked up correctly. Reasons I would rather see it here:

 * Solve it once, why make someone trying to run mock under podman themselves without the toolbox wrapper hit this error and *hopefully* google search and find this PR and find this workaround. (Yes, maybe we could get it changed in the Fedora base containers, but might take a while.)
 * This is really something that is needed for the container we create rather than the enclosing container; e.g., if mock was enhanced at some point to use systemd-nspawn --private-users to avoid needing root privileges, then you'd need the same thing.

That being said, if you really don't want this, I'd rather have the rest of this PR landed :-)